sockets/proxy.go

```diff
@@ -1,51 +0,0 @@
-package sockets
-
-import (
-	"net"
-	"net/url"
-	"os"
-	"strings"
-
-	"golang.org/x/net/proxy"
-)
-
-// GetProxyEnv allows access to the uppercase and the lowercase forms of
-// proxy-related variables.  See the Go specification for details on these
-// variables. https://golang.org/pkg/net/http/
-func GetProxyEnv(key string) string {
-	proxyValue := os.Getenv(strings.ToUpper(key))
-	if proxyValue == "" {
-		return os.Getenv(strings.ToLower(key))
-	}
-	return proxyValue
-}
-
-// DialerFromEnvironment takes in a "direct" *net.Dialer and returns a
-// proxy.Dialer which will route the connections through the proxy using the
-// given dialer.
-func DialerFromEnvironment(direct *net.Dialer) (proxy.Dialer, error) {
```

Comment:
This one as well (likely should be replaced); https://github.com/moby/moby/blob/b0e6eedcf4252d56e62fc3de6de5d01e75a3dc44/distribution/registry.go#L82